### PR TITLE
WIP Switch to iOS 13, use Tasks to handle threads and add error handling

### PR DESCRIPTION
**Pull request by ened**

### CODE REVIEW

File: ios/Classes/MobileScanner.swift [hunk 1]
```diff
@@ -197,36 +197,41 @@ public class MobileScanner: NSObject, AVCaptureVideoDataOutputSampleBufferDelega
...
             }
         }
+        
+        let dimensions = CMVideoFormatDescriptionGetDimensions(self.device.activeFormat.formatDescription)
+        
+        if self.device == nil {
+            throw MobileScannerError.noCamera
+        }
+
+        return MobileScannerStartParameters(
```
Comment (navaronbracke) [inline ios/Classes/MobileScanner.swift:229]: Does this return on the main dispatch queue still?

**Comment by ened** [comment]
Happy to take another look. Closed it due to bug fix already in repo (for the iOS crashes).

**Comment by navaronbracke** [comment]
@ened Is this still on your radar? This might help with fixing another case of #720 occurring (not sure though).

Unfortunately, there have been some changes to the iOS side of the plugin that caused some merge conflicts.
However, adding the `Task` back in for iOS should be trivial.

I can take a look at this as well if needed.

**Comment by juliansteenbakker** [comment]
Hi all, since this PR is stale, i am going to merge it in to one of our branches so we don't alter the origin of @ened. This plugin still needs to support iOS 12, however async does support iOS 12. So i am going to change and test this, and merge if everything works out fine. Source: https://developer.apple.com/documentation/network/nwprotocolframer/instance/3200472-async

Thanks for your work @ened !

**Pull request merged**